workers/price-reporter/src/exchange/connection.rs

```diff
@@ -142,10 +142,19 @@ pub trait ExchangeConnection: Stream<Item = PriceStreamType> + Unpin + Send {
     ) -> Result<Self, ExchangeConnectionError>
     where
         Self: Sized;
+
     /// Send a keepalive signal on the connection if necessary
     async fn send_keepalive(&mut self) -> Result<(), ExchangeConnectionError> {
         Ok(())
     }
+
+    /// Check whether the exchange supports the given pair
+    async fn supports_pair(
+        base_token: &Token,
+        quote_token: &Token,
+    ) -> Result<bool, ExchangeConnectionError>
+    where
+        Self: Sized;
```

Comment:
nit: might be worth just floating the `Sized` trait bound up to the trait signature now that it's on a few methods

Reply:
ah, this makes it impossible for the trait to be object-safe, which we need